edufinder/edufinder/rest_api/views.py

```diff
@@ -68,8 +69,9 @@ def get_nextquestion(previous_answers: List[dict]):
     Expected input format
         [ { id: int, question: str, answer: int }, ... ]
     """
-    return random.choice(list(set([x.id for x in Question.objects.all()]) -
-                              set([x['id'] for x in previous_answers])))
+    used_ids = [x['id'] for x in previous_answers]
+    return Question.objects.exclude(pk__in=used_ids).order_by('-weight').first()
```

Comment:
If all elements have the same weight they should probably be picked randomly not in sequence.
Similar to what @joandrsn said.